packages/terra-menu/src/MenuItem.jsx

```diff
@@ -281,12 +281,12 @@ class MenuItem extends React.Component {
         fitStartIcon = <InstructionsForUseIcon className={cx('start-icon')} />;
       } else if (toggleableMenu) {
         if (!toggleable && icon) {
-          fitStartIcon = React.cloneElement(icon, { className: cx('start-icon') });
+          fitStartIcon = React.cloneElement(icon, { className: `${icon.props.className} ${cx('start-icon')}` });
```

Comment:
Can we have this classes declared outside if block and re-use it where-ever required.
```js
const iconClass = cx([icon.props.className, 'start-icon']);
```
```suggestion
          fitStartIcon = React.cloneElement(icon, { className: iconClass });
```

Reply:
@supreethmr I don't see a good way to accommodate this as the icon may be null outside of where I am appending the classname.  The icon may be null is some cases, therefore, I will always need to have an if check for it, also, in the case of I have an icon and the menu is toggleable, I have an icon given, but we apply the checkmark; therefore, no icon styles are needed but need checkmark classnames.  If I moved the icon classnames out, then I would expect to move the checkmark classnames out as well for when the menu is toggleable or when I need to reserve space.

Therefore, if I moved the declaration of these classes out, I would need to repeat this same if block twice
I feel is is much more clear and much less code to apply it right when the icon is cloned as it is being done currently for the icon and the other start icons already.

Reply:
I think you got it wrong. Let me put it in this way 

This is your code 
```js
        if (!toggleable && icon) {
          fitStartIcon = React.cloneElement(icon, { className: `${icon.props.className} ${cx('start-icon')}` });
        } else {
          fitStartIcon = <CheckIcon className={cx(['checkmark', 'start-icon'])} />;
        }
      } else if (icon) {
        fitStartIcon = React.cloneElement(icon, { className: `${icon.props.className} ${cx('start-icon')}` });
      }
```

Nit: my suggestion was to just take out string concat 
```js
        if (!toggleable && icon) {
          fitStartIcon = React.cloneElement(icon, { className: cx([icon.props.className, 'start-icon']) });
        } else {
          fitStartIcon = <CheckIcon className={cx(['checkmark', 'start-icon'])} />;
        }
      } else if (icon) {
        fitStartIcon = React.cloneElement(icon, { className: cx([icon.props.className, 'start-icon']) });
      }
```
I don't think this would additional if else !!

Reply:
gotcha!  thanks @supreethmr 
I have now updated to match the above